### DISCUSSION TITLE
Issue with nested nuget packages.

### DISCUSSION THREAD

**Discussion by MonicaJa** [NuGet/Home#11233] [Q&A]
Scenario: We have two projects ProjectA and ProjectB. ProjectA is dependent on some C++ libraries and is packed as nuget package. ProjectB is a .net5 class library and ProjectA nuget is added in ProjectB.  Then we create ProjectB  nuget package.
Now if we craete a Console application and add ProjectB nuget, the code complies Ok. When we expand the dependencies section we can see ProjectA under ProjectB, everything looks good. When we run the console application, we get Class not registered error.
The application works fine if we explcitly add ProjectA nuget to the solution.  
Please suggest any solution if you ever experienced any similar issue.
Thank you!

**Comment by nkolev92** [comment]
Hey @MonicaJa, 

Unfortunately I am not familiar with Class not registered error itself. 

Worth trying app referencing ProjectA as a package to see where the runtime problem is actually coming from. 
Referencing native packages sometimes requires that those packages are authored in ways that would get deployed at runtime, but it's all very package dependent.

**Reply by MonicaJa** [reply]
Hi @nkolev92 
Thank you for the reply.
The class not register error comes when the program execution can not find the dll in which the class or method is defined. In our case ProjectA is dependent on VCRTForwarders nuget and when we add ProjectA nuget to ProjecB some dlls are missing, similarly those dlls are missing from ProjectB nuget package. 
As of now the workaround is to add ProjectA nuget whenever we add ProjectB. But the best solution for us is to have all dlls in ProjectB nuget , so that we have to add just one nuget for the applications. 
Please let me know if you need any more information.

Thank you!

**Reply by nkolev92** [reply]
Hi @MonicaJa, 

Unfortunately I'm not super well versed in the specifics for how certain packages such as VCRTForwarders work to help you investigate from that side. 

As far as NuGet is concerned, is it possible that the `build` folders need to be allowed to flow transitively? 
Are you familiar with https://docs.microsoft.com/en-us/nuget/consume-packages/package-references-in-project-files#controlling-dependency-assets

**Reply by MonicaJa** [reply]
Hi @nkolev92 
Thanks for your time. The issue is with the WinRT dlls. We have to explicitly copy the dlls to the parent directory.
